plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseTableProperties.java

```diff
@@ -36,26 +42,120 @@
+                        new ArrayType(VARCHAR),
+                        List.class,
+                        ImmutableList.of(),
+                        false,
+                        value -> ImmutableList.copyOf(((Collection<?>) value).stream()
+                                .map(name -> ((String) name).toLowerCase(ENGLISH))
+                                .collect(Collectors.toList())),
+                        value -> value),
+                new PropertyMetadata<>(
+                        PARTITION_BY_PROPERTY,
+                        "columns to be the partition key. Optional",
+                        new ArrayType(VARCHAR),
+                        List.class,
+                        ImmutableList.of(),
+                        false,
+                        value -> ImmutableList.copyOf(((Collection<?>) value)),
+                        value -> value),
+                new PropertyMetadata<>(
+                        PRIMARY_KEY_PROPERTY,
+                        "columns to be the primary key. Optional",
+                        new ArrayType(VARCHAR),
+                        List.class,
+                        ImmutableList.of(),
+                        false,
+                        value -> ImmutableList.copyOf(((Collection<?>) value).stream()
+                                .map(name -> ((String) name).toLowerCase(ENGLISH))
+                                .collect(Collectors.toList())),
+                        value -> value),
+                stringProperty(
+                        SAMPLE_BY_PROPERTY,
+                        "An expression for sampling. Optional.",
+                        null,
                         false));
     }
 
-    public static Optional<String> getEngine(Map<String, Object> tableProperties)
+    public static ClickHouseEngineType getEngine(Map<String, Object> tableProperties)
+    {
+        requireNonNull(tableProperties);
+        return (ClickHouseEngineType) tableProperties.get(ENGINE_PROPERTY);
+    }
+
+    @SuppressWarnings("unchecked")
+    public static Optional<String> getOrderBy(Map<String, Object> tableProperties)
+    {
+        requireNonNull(tableProperties);
+        List<String> orderBy = (List<String>) tableProperties.get(ORDER_BY_PROPERTY);
+        return formatProperty(orderBy);
+    }
+
+    @SuppressWarnings("unchecked")
+    public static Optional<String> getPartitionBy(Map<String, Object> tableProperties)
     {
         requireNonNull(tableProperties);
+        List<String> partitionBy = (List<String>) tableProperties.get(PARTITION_BY_PROPERTY);
+        return formatProperty(partitionBy);
+    }
+
+    @SuppressWarnings("unchecked")
+    public static Optional<String> getPrimaryKey(Map<String, Object> tableProperties)
+    {
+        requireNonNull(tableProperties);
+        List<String> primaryKey = (List<String>) tableProperties.get(PRIMARY_KEY_PROPERTY);
+        return formatProperty(primaryKey);
```

Comment:
The formatting should not happen here. This method should return the session property as-is.

Please maybe make `formatProperty` public and call it from `ClickHouseClient`

Reply:
OK, I move it to `ClickHouseClient` class